<?xml version="1.0"?>
<!--
Launch the path follower with setup for the selected robot controller.

To specify which robot controller/model is used, set the name of the controller in the
environment variable `ROBOT_CONTROLLER` or set the parameter externally. Default is 'ackermann_purepursuit'.
Default parameters for all controllers are loaded here as well.
Please make sure, you add an according launch file, if you implement a new controller.
-->
<launch>
    <node pkg="path_follower" type="path_follower_node" name="path_follower" output="screen" respawn="true">
        <param name="controller_type" value="$(optenv ROBOT_CONTROLLER ackermann_purepursuit)" />
    </node>

    <!-- default settings for the follower -->
    <group ns="path_follower">
        <param name="min_velocity" value="0.5" />
        <param name="max_velocity" value="2.5" />
        <param name="waypoint_tolerance" value="0.8" />
        <param name="goal_tolerance" value="0.1" />

        <group ns="collision_avoider">
            <group ns="collision_box">
                <param name="width" value="0.4" />
                <param name="crit_length" value="0.6" />
                <param name="min_length" value="1.0" />
                <param name="max_length" value="1.5" />
                <param name="velocity_factor" value="1.0" />
                <param name="velocity_saturation" value="1.0" />
            </group>
        </group>

        <group ns="supervisor">
            <param name="use_waypoint_timeout" value="false" />

            <param name="use_distance_to_path" value="true" />
            <group ns="distance_to_path">
                <param name="max_dist" value="2.0" />
            </group>

            <param name="use_path_lookout" value="true" />
            <group ns="path_lookout">
                <param name="obstacle_scale_distance" value="1.0" />
                <param name="obstacle_scale_lifetime" value="5.0" />
                <param name="path_width" value="0.5" />
            </group>
        </group>
    </group>

    <!-- load common parameters -->
    <include file="$(find path_follower)/launch/follower_2steer_inputscaling.launch" />
    <include file="$(find path_follower)/launch/follower_2steer_purepursuit.launch" />
    <include file="$(find path_follower)/launch/follower_2steer_stanley.launch" />
    <include file="$(find path_follower)/launch/follower_ackermann_inputscaling.launch" />
    <include file="$(find path_follower)/launch/follower_ackermann_orthexp.launch" />
    <include file="$(find path_follower)/launch/follower_ackermann_pid.launch" />
    <include file="$(find path_follower)/launch/follower_ackermann_purepursuit.launch" />
    <include file="$(find path_follower)/launch/follower_ackermann_stanley.launch" />
    <include file="$(find path_follower)/launch/follower_differential_orthexp.launch" />
    <include file="$(find path_follower)/launch/follower_dynamic_SLP.launch" />
    <include file="$(find path_follower)/launch/follower_dynamic_window.launch" />
    <include file="$(find path_follower)/launch/follower_ICR_CCW.launch" />
    <include file="$(find path_follower)/launch/follower_kinematic_HBZ.launch" />
    <include file="$(find path_follower)/launch/follower_kinematic_HBZ_TT.launch" />
    <include file="$(find path_follower)/launch/follower_kinematic_SLP.launch" />
    <include file="$(find path_follower)/launch/follower_OFC.launch" />
    <include file="$(find path_follower)/launch/follower_omnidrive_orthexp.launch" />
    <include file="$(find path_follower)/launch/follower_potential_field.launch" />
    <include file="$(find path_follower)/launch/follower_potential_field_TT.launch" />
    <include file="$(find path_follower)/launch/follower_scitos_orthexp.launch" />
    <include file="$(find path_follower)/launch/follower_unicycle_inputscaling.launch" />

</launch>
